### PR TITLE
Refs #6280 Added history_depth option. [6295]

### PR DESCRIPTION
**Pull request by LuisGP**
Async test may remove from history (default depth 1) samples before the writer's thread retrieve them.
Added an option to allow chaging history_depth value, and modified AsyncRTPSAsReliableWithPersistence test to use it.

### CODE REVIEW

File: test/blackbox/BlackboxTestsPersistence.cpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ TEST_F(BlackBoxPersistence, AsyncRTPSAsReliableWithPersistence)
 
     ASSERT_TRUE(reader.isInitialized());
 
-    writer.make_persistent(db_file_name(), guid_prefix()).asynchronously(eprosima::fastrtps::rtps::RTPSWriterPublishMode::ASYNCHRONOUS_WRITER).init();
+    writer.make_persistent(db_file_name(), guid_prefix()).history_depth(10).asynchronously(eprosima::fastrtps::rtps::RTPSWriterPublishMode::ASYNCHRONOUS_WRITER).init();
```
Comment (MiguelCompany) [inline test/blackbox/BlackboxTestsPersistence.cpp]: This line is too long

File: test/blackbox/RTPSWithRegistrationWriter.hpp [hunk 2]
```diff
@@ -142,9 +142,9 @@ class RTPSWithRegistrationWriter
 
         while(it != msgs.end())
         {
-	    eprosima::fastrtps::rtps::CacheChange_t * ch = writer_->new_change(*it,eprosima::fastrtps::rtps::ALIVE);
+        eprosima::fastrtps::rtps::CacheChange_t * ch = writer_->new_change(*it,eprosima::fastrtps::rtps::ALIVE);
```
Comment (MiguelCompany) [inline test/blackbox/RTPSWithRegistrationWriter.hpp]: 4 more spaces

File: test/blackbox/RTPSWithRegistrationWriter.hpp [hunk 3]
```diff
@@ -142,9 +142,9 @@ class RTPSWithRegistrationWriter
...
 
-	    eprosima::fastcdr::FastBuffer buffer((char*)ch->serializedPayload.data, ch->serializedPayload.max_size);
+        eprosima::fastcdr::FastBuffer buffer((char*)ch->serializedPayload.data, ch->serializedPayload.max_size);
```
Comment (MiguelCompany) [inline test/blackbox/RTPSWithRegistrationWriter.hpp]: 4 more spaces

File: test/blackbox/RTPSWithRegistrationWriter.hpp [hunk 4]
```diff
@@ -180,8 +180,8 @@ class RTPSWithRegistrationWriter
     /*** Function to change QoS ***/
     RTPSWithRegistrationWriter& memoryMode(const eprosima::fastrtps::rtps::MemoryManagementPolicy_t memoryPolicy)
     {
-	hattr_.memoryPolicy = memoryPolicy;
-	return *this;
+    hattr_.memoryPolicy = memoryPolicy;
```
Comment (MiguelCompany) [inline test/blackbox/RTPSWithRegistrationWriter.hpp]: 4 more spaces on both lines

File: test/blackbox/RTPSWithRegistrationWriter.hpp [hunk 5]
```diff
@@ -250,6 +250,12 @@ class RTPSWithRegistrationWriter
             .add_property("dds.persistence.sqlite3.filename", filename);
     }
 
+    RTPSWithRegistrationWriter& history_depth(const int32_t depth)
```
Comment (MiguelCompany) [inline test/blackbox/RTPSWithRegistrationWriter.hpp]: Parameter on next line

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/376/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/342/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/374/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/396/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/361/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/394/)

**Pull request merged**